src/pages-and-resources/discussions/app-list/AppList.jsx

```diff
@@ -42,8 +55,49 @@ const AppList = ({ intl }) => {
     dispatch(updateValidationStatus({ hasError: false }));
   }, [selectedAppId, activeAppId]);
 
+  useEffect(() => {
+    setDiscussionEnabled(enabled);
+  }, [enabled]);
+
+  useEffect(() => {
+    dispatch(fetchDiscussionSettings(courseId, selectedAppId));
+  }, [courseId, selectedAppId]);
```

Comment:
It will call on every provider selection. It should send Call only once because enabled and postingRestrictions response are the same regardless of provider.

Reply:
It will stop unnecessary calls when the user will move forward and backward.
useEffect(() => {
    if (!postingRestrictions) {
      dispatch(fetchDiscussionSettings(courseId, selectedAppId));
    }
  }, [courseId]);